### PR TITLE
Fix failed tests for branch-2.10

### PR DESCRIPTION
**Pull request by codelipenghui**
Fixes: #16000

After cherry-picked #15627, the test `org.apache.bookkeeper.mledger.impl.ManagedLedgerTest`
will fail for branch-2.10. Details to see https://github.com/apache/pulsar/pull/15627#issuecomment-1149962309

### CODE REVIEW

**Review by nicoloboschi — APPROVED** [review]

**Review by RobertIndie — APPROVED** [review]

File: managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java [hunk 1]
```diff
@@ -2293,8 +2293,8 @@ public void testGetNumberOfEntriesInStorage() throws Exception {
         stateUpdater.set(managedLedger, ManagedLedgerImpl.State.LedgerOpened);
         managedLedger.rollCurrentLedgerIfFull();
         Awaitility.await().untilAsserted(() -> {
-            assertEquals(managedLedger.getLedgersInfo().size(), 2);
-            assertEquals(managedLedger.getState(), ManagedLedgerImpl.State.ClosedLedger);
+            assertEquals(managedLedger.getLedgersInfo().size(), 3);
```
Comment (gaoran10) [inline managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java:2296]: It seems that this changed the test logic.

Comment (codelipenghui) [reply managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java:2296]: Yes, please see the description of this PR.

**Review by congbobo184 — APPROVED** [review]

**Review by Demogorgon314 — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by eolivelli — COMMENTED** [review]
@codelipenghui can you please add in the description the reason why we need to set -DtestReuseFork=false ?

**Comment by codelipenghui** [comment]
> @codelipenghui can you please add in the description the reason why we need to set -DtestReuseFork=false ?

Not 100% sure what's going on for now, but looks like It's related to the concurrency issue in the class initialization. Just found some related information. 

https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html#parallel-test-execution

https://github.com/testcontainers/testcontainers-java/issues/2939#issuecomment-649810918

**Pull request merged**